test/e2e/apimachinery/apply.go

```diff
@@ -0,0 +1,1181 @@
+				for _, c := range pod.Status.Conditions {
+					if c.Type == "MyStatus" {
+						myStatusFound = true
+						break
+					}
+				}
+				if myStatusFound == false {
+					framework.Failf("Expected pod to have applied status")
+				}
+			}
+		}
+	})
+
+	/*
+		Release : v1.21
+		Testname: Server Side Apply, unset field
+		Description: Apply an object. Issue a subsequent apply that removes a field. The particular field MUST be removed.
+	*/
+	ginkgo.It("should remove a field if it is owned but removed in the apply request", func() {
+		obj := []byte(`{
+		"apiVersion": "apps/v1",
+		"kind": "Deployment",
+		"metadata": {
+			"name": "deployment",
+			"labels": {"app": "nginx"}
+		},
+		"spec": {
+			"replicas": 3,
+			"selector": {
+				"matchLabels": {
+					"app": "nginx"
+				}
+			},
+			"template": {
+				"metadata": {
+					"labels": {
+						"app": "nginx"
+					}
+				},
+				"spec": {
+					"containers": [{
+						"name":  "nginx",
+						"image": "nginx:latest",
+						"ports": [{
+							"containerPort": 80,
+							"protocol": "TCP"
+						}]
+					}]
+				}
+			}
+		}
+	}`)
+
+		_, err := client.CoreV1().RESTClient().Patch(types.ApplyPatchType).
+			AbsPath("/apis/apps/v1").
+			Namespace(ns).
+			Resource("deployments").
+			Name("deployment").
+			Param("fieldManager", "apply_test").
+			Body(obj).Do(context.TODO()).Get()
+		if err != nil {
+			framework.Failf("Failed to create object using Apply patch: %v", err)
+		}
+
+		obj = []byte(`{
+		"apiVersion": "apps/v1",
+		"kind": "Deployment",
+		"metadata": {
+			"name": "deployment",
+			"labels": {"app": "nginx"}
+		},
+		"spec": {
+			"replicas": 3,
+			"selector": {
+				"matchLabels": {
+					"app": "nginx"
+				}
+			},
+			"template": {
+				"metadata": {
+					"labels": {
+						"app": "nginx"
+					}
+				},
+				"spec": {
+					"containers": [{
+						"name":  "nginx",
+						"image": "nginx:latest"
+					}]
+				}
+			}
+		}
+	}`)
+
+		_, err = client.CoreV1().RESTClient().Patch(types.ApplyPatchType).
+			AbsPath("/apis/apps/v1").
+			Namespace(ns).
+			Resource("deployments").
+			Name("deployment").
+			Param("fieldManager", "apply_test").
+			Body(obj).Do(context.TODO()).Get()
+		if err != nil {
+			framework.Failf("Failed to remove container port using Apply patch: %v", err)
+		}
+
+		deployment, err := client.AppsV1().Deployments(ns).Get(context.TODO(), "deployment", metav1.GetOptions{})
+		if err != nil {
+			framework.Failf("Failed to retrieve object: %v", err)
+		}
+
+		if len(deployment.Spec.Template.Spec.Containers[0].Ports) > 0 {
+			framework.Failf("Expected no container ports but got: %v, object: \n%#v", deployment.Spec.Template.Spec.Containers[0].Ports, deployment)
+		}
+
+	})
+
+	/*
+		Release : v1.21
+		Testname: Server Side Apply, unset field shared
+		Description: Apply an object. Unset ownership of a field that is also owned by other managers and make a subsequent apply request. The unset field MUST not be removed from the object.
```

Comment:
We could verify the managed fields are what we expect as well to ensure the ownership transfer really worked.

Reply:
It's fairly verbose the check the managedFields without importing SMD. I think it should be enough that removing the field from the first applier didn't delete the field to recognize that the field is still owned by the other applier.

Reply:
I agree. Thanks for checking.